### DISCUSSION TITLE
How do I call an analytics function every time the page is loaded?

### DISCUSSION THREAD

**Discussion by daydream05** [gatsbyjs/gatsby#27794] [Help]
Also, I would need the data from the page that is rendered.

I'm looking at gatsby-browser apis but not sure which one to use
```
exports.onClientEntry = () => {
  console.log("We've started!")
  callAnalyticsAPI()
}
```
onClientEntry doesn't have data from the page

I was using `useEffect(() =>, [])` before but sometimes the analytics hasn't loaded yet which causes referencerror.

**Comment by herecydev** [comment]
[WrapPageElement](https://www.gatsbyjs.com/docs/browser-apis/#wrapPageElement) will give you access to the pageProps. I would suggest using a `useEffect` in there. I'm unaware of your analytics provider so it's not obvious how to solve the referenceerror issue.

**Reply by daydream05** [reply]
Thanks! I use [amplitude](https://amplitude.com/) and the [gatsby plugin ](https://www.gatsbyjs.com/plugins/gatsby-plugin-amplitude-analytics/)for it

  // For analytics
```
  useEffect(() => {
    if (typeof window !== `undefined`) {
       logAmplitudeEvent({ eventName: `Docs Topic: ${documentation?.title}` })
    }
  }, [])
```
